### PR TITLE
Emails: Add to the JITM modal the possibility to show a different image based on the icon property

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Proposed Changes

* Allows the JITM modal to change its sidebar image

#### Testing Instructions

1. It needs to be done by applying the patch D83166-code (and following its instructions)
2. Assert that the image is as follows
3. Test that the previous JITM keeps working as before (We don't provide icon property)

For Embedded Inbox launch
image

For plans (Not providing icon property in the JITM engine
image

Related to 1200182182542585-as-1202476537751009

### CODE REVIEW

File: client/blocks/jitm/templates/modal.jsx [hunk 1]
```diff
@@ -1,24 +1,48 @@
...
+				return Visual;
+		}
+	};
+
+	const getModalAltText = () => {
+		switch ( icon ) {
+			case 'mailbox':
+				return 'Embedded inbox';
+			default:
+				return 'list of available plans — premium, business, and ecommerce plans';
```
Comment (stephanethomas) [inline client/blocks/jitm/templates/modal.jsx]: Those strings should be localized (I suspect they only appear in screen readers).

**Comment by stephanethomas** [comment]
Unfortunately there is an issue with this image - it's in English:

image

That's not an easy thing to fix though, as I don't think generating images for every language we support is going to work.

/cc @semihakocer

**Review by stephanethomas — CHANGES_REQUESTED** [review]
Putting aside the image which is not localized, I've also noticed this weird bug where the cross in the modal would be highlighted when the JITM appears, and its tooltip would be cropped (I'm on Windows with Chrome if that helps):



It's unrelated to your changes but we may want to escalate this issue to Marketing.

File: client/blocks/jitm/templates/modal.jsx (hunk 1 above)
Comment (AllTerrainDeveloper) [reply client/blocks/jitm/templates/modal.jsx]: Yes, only for screen readers. Addressed

**Comment by AllTerrainDeveloper** [comment]
@stephanethomas Thank you! I've addressed your concerns and I'm pinging the team in order to warn them about the close text in the modal

**Comment by AllTerrainDeveloper** [comment]
> Unfortunately there is an issue with this image - it's in English:
> 
> image
> 
> That's not an easy thing to fix though, as I don't think generating images for every language we support is going to work.
> 
> /cc @semihakocer

We've decided to target for now only English speakers. Thanks for catching that too!

File: client/blocks/jitm/templates/modal.jsx [hunk 2]
```diff
@@ -1,23 +1,60 @@
...
+				return Visual;
+		}
+	};
+
+	const getModalAltText = () => {
+		switch ( icon ) {
+			case 'embedded-inbox':
+				return translate( 'Embedded inbox', { textOnly: true } );
+			default:
+				return translate( 'list of available plans — premium, business, and ecommerce plans', {
```
Comment (stephanethomas) [inline client/blocks/jitm/templates/modal.jsx]: I would suggest the following text, so we don't have to update it when we launch new plans (it's already stale 😁):
```suggestion
				return translate( 'List of available WordPress.com plans', {
```

File: client/blocks/jitm/templates/modal.jsx [hunk 3]
```diff
@@ -1,23 +1,60 @@
 import { Button, Guide } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
 import { useState } from 'react';
 import './modal-style.scss';
+import Inbox from './inbox.svg';
```
Comment (stephanethomas) [inline client/blocks/jitm/templates/modal.jsx]: Thinking about it more, we could name this file `embedded-inbox` so it is less generic and more consistent with the name of the icon.

File: client/blocks/jitm/templates/modal.jsx [hunk 4]
```diff
@@ -1,23 +1,60 @@
...
+			case 'embedded-inbox':
+				return translate( 'Embedded inbox', { textOnly: true } );
+			default:
+				return translate( 'list of available plans — premium, business, and ecommerce plans', {
+					textOnly: true,
+				} );
+		}
+	};
+
+	const getClassName = () => {
```
Comment (stephanethomas) [inline client/blocks/jitm/templates/modal.jsx]: We may want to name this `getModalClassName()` to be more consistent with `getModalImage()` and `getModalAltText()`.

File: client/blocks/jitm/templates/modal.jsx [hunk 5]
```diff
@@ -53,12 +90,7 @@ export default function ModalTemplate( {
 									<p className="modal__disclaimer">{ line }</p>
 								) ) }
 							</div>
-							<div className="modal__sidebar">
-								<img
-									src={ Visual }
-									alt="list of available plans — premium, business, and ecommerce plans"
-								/>
-							</div>
+							<div className="modal__sidebar">{ modalImage }</div>
```
Comment (stephanethomas) [inline client/blocks/jitm/templates/modal.jsx]: This variable doesn't look necessary, we could inline everything:

```suggestion
							<div className="modal__sidebar">
								<img className={ getClassName() } src={ getModalImage() } alt={ getModalAltText() } />
							</div>
```

**Review by stephanethomas — APPROVED** [review]
I had a number of minor suggestions but otherwise it's all good!

**Comment by AllTerrainDeveloper** [comment]
@stephanethomas I've added all your suggestions :)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7404182

Thank you @AllTerrainDeveloper for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7404182).

**Pull request merged**